addons/account/views/report_payment_receipt_templates.xml

```diff
@@ -52,22 +53,28 @@
                                     <td><span t-field="inv.invoice_date"/></td>
                                     <td><span t-field="inv.name"/></td>
                                     <td><span t-field="inv.ref"/></td>
+                                    <td/>
                                     <td class="text-right"><span t-field="inv.amount_total"/></td>
                                 </tr>
                                 <!-- PAYMENTS/REVERSALS -->
                                 <tr t-foreach="inv._get_reconciled_invoices_partials()" t-as="par">
-                                    <t t-set="amount" t-value="par[1]"/>
+                                    <t t-set="amountPayment" t-value="par[2].amount_currency"/>
+                                    <t t-set="currencyPayment" t-value="par[0].credit_currency_id"/>
+                                    <t t-set="amountInvoice" t-value="par[1]"/>
+                                    <t t-set="currencyInvoice" t-value="par[0].debit_currency_id"/>
                                     <t t-set="payment" t-value="par[2].move_id"/>
                                     <td><span t-field="payment.date"/></td>
                                     <td><span t-field="payment.name"/></td>
                                     <td><span t-field="payment.ref"/></td>
-                                    <td class="text-right">-<span t-esc="amount" t-options="{'widget': 'monetary', 'display_currency': o.currency_id}"/></td>
+                                    <td class="text-right"><span t-esc="amountPayment" t-options="{'widget': 'monetary', 'display_currency': currencyPayment}"/></td>
+                                    <td class="text-right">-<span t-esc="amountInvoice" t-options="{'widget': 'monetary', 'display_currency': currencyInvoice}"/></td>
```

Comment:
The explicit `-` may be rendered before the currency sign which is not ideal. 
Ideally we would need `amountPayment` and `amountInvoice` already with the correct sign